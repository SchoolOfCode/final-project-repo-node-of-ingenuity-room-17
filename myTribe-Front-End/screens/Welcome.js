import {
  StyleSheet,
  Text,
  View,
  ScrollView,
  ImageBackground,
} from "react-native";
import UserThumbnail from "../components/UserThumbnail";
import { useState, useEffect, useContext } from "react";
import headerImage from "../assets/welcome-background.jpeg";
import {pageState} from "../App"

const Welcome = (props) => {
  const {family, setFamily} = useContext(pageState)

  // useEffect(() => {
  //   const userFamily = props.navigation.getParam("family");
  //   setFamily(userFamily);
  // }, []);

  const selectHandler = (name) => {
    setFamily({...family, memberSession: name})
    props.navigation.navigate({ routeName: "Dashboard" });
  };
  return (
    <View style={styles.container}>
      <ScrollView contentContainerStyle={styles.scrollView}>
        <ImageBackground
          resizeMode="cover"
          style={styles.header}
          source={headerImage}
        >
          <View style={styles.overlay}>
            <Text style={styles.heading}>Welcome back</Text>
            <Text style={styles.subHeading}>
              Start taking control of your household.
            </Text>
          </View>
        </ImageBackground>
        <View style={styles.profiles}>
          <Text style={styles.sectionHeading}>Please select your account</Text>
          {family &&
            family.members.map((el) => (
              <UserThumbnail
                key={el.id}
                name={el.name}
                isParent={el.isParent}
                selectHandler={selectHandler}
              />
            ))}
        </View>
      </ScrollView>
    </View>
  );
};
const styles = StyleSheet.create({
  container: {
    height: "100%",
    width: "100%",
    justifyContent: "space-around",
  },

  scrollView: {
    width: "100%",
    height: "100%",
    backgroundColor: "#F2F0F0",
  },

  header: {
    height: 250,
    width: "100%",
    resizeMode: "cover",
  },

  overlay: {
    justifyContent: "center",
    height: "100%",
    width: "100%",
    backgroundColor: "rgba(0,0,0,0.3)",
    paddingLeft: 50,
    paddingRight: 50,
  },

  heading: {
    fontSize: 40,
    fontWeight: "bold",
    color: "white",
  },

  subHeading: {
    fontSize: 20,
    color: "white",
  },
  profiles: {
    padding: 50,
    flexDirection: "row",
    flexWrap: "wrap",
    alignItems: "center",
    alignContent: "center",
    justifyContent: "space-evenly",
  },
  sectionHeading: {
    fontSize: 24,
    fontWeight: "bold",
    marginBottom: 50,
  },
});
export default Welcome;
