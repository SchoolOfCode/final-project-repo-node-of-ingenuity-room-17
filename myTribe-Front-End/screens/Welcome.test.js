import React from "react";
import renderer from "react-test-renderer";

import Welcome from "./Dashboard";

describe("<Welcome />", () => {
  it("has 2 children", () => {
    const tree = renderer.create(<Welcome />).toJSON();
    expect(tree.children.length).toBe(2);
  });
  it("renders correctly", () => {
    const tree = renderer.create(<Welcome />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
