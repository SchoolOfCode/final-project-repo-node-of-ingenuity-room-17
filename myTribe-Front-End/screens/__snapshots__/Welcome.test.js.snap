// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Welcome /> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "height": 250,
        "position": "relative",
        "resizeMode": "cover",
        "width": "100%",
      }
    }
  >
    <Image
      resizeMode="cover"
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 250,
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,0.3)",
          "height": "100%",
          "justifyContent": "center",
          "paddingLeft": 50,
          "paddingRight": 50,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 40,
            "fontWeight": "bold",
          }
        }
      >
        Dashboard
      </Text>
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "marginTop": 40,
          }
        }
      >
        Emma
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FEB800",
            "borderTopLeftRadius": 10,
            "borderTopRightRadius": 10,
            "flexDirection": "row",
            "height": "50%",
            "justifyContent": "space-evenly",
            "top": -18,
            "width": "100%",
          },
          Object {
            "zIndex": 2,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 10,
            "elevation": 5,
            "height": "77%",
            "justifyContent": "center",
            "marginTop": -8,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
            "width": "30%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 20,
              "fontWeight": "bold",
              "paddingTop": 20,
              "textAlign": "center",
            }
          }
        >
          Chores
        </Text>
        <Text
          style={
            Object {
              "color": "#FEB800",
              "fontSize": 40,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          5
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 10,
            "elevation": 5,
            "height": "77%",
            "justifyContent": "center",
            "marginTop": -8,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
            "width": "30%",
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": "70%",
              "width": "80%",
            }
          }
        />
      </View>
    </View>
  </View>
  <RCTScrollView
    style={
      Object {
        "backgroundColor": "#F2F0F0",
        "borderRadius": 10,
        "color": "black",
        "height": "100%",
        "textAlign": "center",
        "top": 100,
        "width": "100%",
        "zIndex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "color": "white",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "justifyContent": "space-evenly",
            "marginTop": 50,
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "white",
                "borderRadius": 15,
                "elevation": 5,
                "flexDirection": "column",
                "height": 130,
                "justifyContent": "center",
                "margin": 20,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
                "width": 150,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "alignSelf": "center",
                  "height": 60,
                  "width": 60,
                }
              }
            />
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 18,
                  "padding": 5,
                  "top": 10,
                }
              }
            >
              View Chores
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "white",
                "borderRadius": 15,
                "elevation": 5,
                "flexDirection": "column",
                "height": 130,
                "justifyContent": "center",
                "margin": 20,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
                "width": 150,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "alignSelf": "center",
                  "height": 60,
                  "width": 60,
                }
              }
            />
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 18,
                  "padding": 5,
                  "top": 10,
                }
              }
            >
              Manage Chores
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "white",
                "borderRadius": 15,
                "elevation": 5,
                "flexDirection": "column",
                "height": 130,
                "justifyContent": "center",
                "margin": 20,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
                "width": 150,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "alignSelf": "center",
                  "height": 60,
                  "width": 60,
                }
              }
            />
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 18,
                  "padding": 5,
                  "top": 10,
                }
              }
            >
              Manage Family
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 15,
              "elevation": 5,
              "flexDirection": "column",
              "height": 130,
              "justifyContent": "center",
              "margin": 20,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3.84,
              "width": 150,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "alignSelf": "center",
                "height": 60,
                "width": 60,
              }
            }
          />
          <Text
            style={
              Object {
                "alignSelf": "center",
                "fontSize": 18,
                "padding": 5,
                "top": 10,
              }
            }
          >
            Show Streak
          </Text>
        </View>
      </View>
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#FEB800",
                },
              ]
            }
          >
            Log Out
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
